src/pybind/mgr/balancer/module.py

```diff
@@ -1144,14 +1143,16 @@ def do_read_balancing(self, plan: Plan) -> Tuple[int, str]:
             self.no_optimization_needed = True
             self.log.debug('unable to balance reads.')
             return -errno.EALREADY, msg
-        self.log.info('prepared {} read changes'.format(total_num_changes))
+        if total_num_changes != 0:
+            self.log.info('prepared {} read changes'.format(total_num_changes))
         if total_num_changes == 0:
+            self.log.debug('prepared {} read changes'.format(total_num_changes))
             self.no_optimization_needed = True
             return -errno.EALREADY, msg
```

Comment:
From lines 1142-1151, I think it would be cleaner to do this:

```
if total_num_changes > 0:
    self.log.info('prepared {} read changes'.format(total_num_changes))
else:
    self.log.debug('prepared {} read changes'.format(total_num_changes))
    self.no_optimization_needed = True
    return -errno.EALREADY, msg
```

When I originally wrote this, I differentiated between a negative number of changes and 0 changes. But, since it should be impossible to ever hit negative changes (we should still log it though if it somehow happens), I think it is better to combine it with the case of 0 total changes. In both cases, we set `no_optimization_needed=True`. And the error message applies to both cases.